torch/csrc/api/include/torch/nn/functional/vision.h

```diff
@@ -46,6 +47,53 @@ inline Tensor affine_grid(
   return torch::affine_grid_generator(theta, size, align_corners);
 }
 
+inline Tensor grid_sample(
+    const Tensor& input,
+    const Tensor& grid,
+    GridSampleOptions options = {}) {
+
+  if ((options.mode().compare("bilinear") != 0) && (options.mode().compare("nearest") != 0)) {
+    TORCH_CHECK(false, "nn::functional::grid_sample(): expected mode to be ",
+                         "'bilinear' or 'nearest', but got: '", options.mode(), "'");
+  }
+
+  if ((options.padding_mode().compare("zeros") != 0) && 
+      (options.padding_mode().compare("border") != 0) && 
+      (options.padding_mode().compare("reflection") != 0)) {
+    TORCH_CHECK(false, "nn.functional.grid_sample(): expected padding_mode ",
+                         "to be 'zeros', 'border', or 'reflection', ",
+                         "but got: '", options.padding_mode(), "'");
+  }
+
+  int64_t mode_enum, padding_mode_enum;
+
+  if (options.mode().compare("bilinear") == 0) {
+    mode_enum = 0;
+  }
+  else { /// mode == 'nearest'
+    mode_enum = 1;
+  }
+
+  if (options.padding_mode().compare("zeros") == 0) {
+    padding_mode_enum = 0;
+  }
+  else if (options.padding_mode().compare("border") == 0) {
+    padding_mode_enum = 1;
+  }
+  else { /// padding_mode == 'reflection'
+    padding_mode_enum = 2;
+  }
+  
+  if (!options.align_corners().has_value()) {
+    TORCH_WARN("Default grid_sample and affine_grid behavior will be changed ",
+                "to align_corners=False from 1.4.0. ",
+                "See the documentation of grid_sample for details.");
+    options.align_corners(true);
```

Comment:
Curious is there a reason why the warning message and the `options.align_corners(true)` setting different from the Python version?

Reply:
Sry, i saw the old 1.3 doc
https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#grid_sample
i will fix it!